src/components/ListItem.jsx

```diff
@@ -1,5 +1,45 @@
+import { useEffect, useState } from 'react';
+import { updateItem } from '../api/firebase';
 import './ListItem.css';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ item, listToken }) {
+	const [isChecked, setIsChecked] = useState(false);
+
+	const { name, id, totalPurchases, dateLastPurchased } = item;
+
+	const handleChange = async (e) => {
+		const { checked } = e.target;
+
+		if (checked) {
+			try {
+				await updateItem(listToken, id, totalPurchases);
+				console.log('Shopping item successfully updated');
+			} catch (error) {
+				console.log(error);
+			}
+		}
+	};
+
+	useEffect(() => {
+		if (dateLastPurchased == null) return;
+
+		const date = dateLastPurchased.toMillis();
+		const twentyFourHrs = 24 * 60 * 60 * 1000;
+
+		const hasElapsed = Date.now() - date <= twentyFourHrs;
+
+		setIsChecked(hasElapsed);
+	});
```

Comment:
you are missing a dependencies array here for the `useEffect`hook. It may cause an infinite loop.